cibuildwheel/util.py

```diff
@@ -2,19 +2,26 @@
 import platform as platform_module
 import re
 import ssl
+import sys
+import functools
 import textwrap
 import urllib.request
 from enum import Enum
 from fnmatch import fnmatch
 from pathlib import Path
 from time import sleep
-from typing import Dict, List, NamedTuple, Optional
+from typing import Dict, List, NamedTuple, Optional, Set
 
 import certifi
 
 from .environment import ParsedEnvironment
 from .typing import PathOrStr
 
+if sys.version_info < (3, 8):
+    from typing_extensions import Literal
```

Comment:
Ah, that's handy

Reply:
Everything that goes into typing that can be backported goes here. :)